libs/community/langchain_community/chat_message_histories/cratedb.py

```diff
@@ -0,0 +1,109 @@
+import json
+import typing as t
+
+import sqlalchemy as sa
+from langchain.schema import BaseMessage, _message_to_dict, messages_from_dict
+
+from langchain_community.chat_message_histories.sql import (
+    BaseMessageConverter,
+    SQLChatMessageHistory,
+)
+
+
+def create_message_model(table_name, DynamicBase):  # type: ignore
+    """
+    Create a message model for a given table name.
+
+    This is a specialized version for CrateDB for generating integer-based primary keys.
+    TODO: Find a way to converge CrateDB's generate_random_uuid() into a variant
```

Comment:
Can you do something like:

```python
b = bytearray()
b.extend(map(ord, uuid))
i = int.from_bytes(b, "big")
```

Reply:
Thanks for spotting this. I think the TODO comment is from a previous version of the code. I think the canonical form to define primary keys with SQLAlchemy and CrateDB to synthesize the almost-equivalent of "autoincrement" columns is just that how it is written down here:

```python
sa.Column(sa.BigInteger, primary_key=True, server_default=sa.func.now())
```

FYI, FWIW: If you can't afford to define/override the SQLAlchemy ORM model definitions, but have to support a column definition using `primary_key=True, autoincrement=True`, and can't change it, there is a support method mentioned at [Synthetic Autoincrement using Timestamps](https://cratedb.com/docs/sqlalchemy-cratedb/support.html#synthetic-autoincrement-using-timestamps).

Reply:
I've just removed the inline TODO comment. Thanks again.